src/utils/auto-box-collider.js

```diff
@@ -1,13 +1,70 @@
+function isVisibleUpToRoot(node, root) {
+  let child = node;
+  while (child) {
+    if (child === root) return true;
+    if (!child.visible) return false;
+    child = child.parent;
+  }
+  console.error(`Root ${root} is not in the hierarchy of node ${node}`);
```

Comment:
"in the hierarchy of" to me reads like it should be a child even though I think its technically correct. I might be more specific and say "is not an ancestor of"